apis/python/operator/src/lib.rs

```diff
@@ -81,30 +84,18 @@ impl PyEvent {
 
     /// Returns the payload of an input event as an arrow array (if any).
     fn value(&self, py: Python<'_>) -> PyResult<Option<PyObject>> {
-        if let Some(data) = &self.data {
-            let data_type = match &self.event {
-                MergedEvent::Dora(Event::Input { metadata, .. }) => match &metadata
-                    .schema
-                    .fields()
-                    .first()
-                    .context("no field in schema")?
-                    .data_type()
-                {
-                    DataType::List(field) => field.data_type().clone(),
-                    _ => todo!(),
-                },
-                _ => DataType::UInt8,
-            };
-            let array = data
-                .clone()
-                .into_arrow_array(data_type)
-                .map_err(|err| arrow::pyarrow::PyArrowException::new_err(err.to_string()))?;
-            // TODO: Does this call leak data?
-            let array_data = array.to_pyarrow(py)?;
-            return Ok(Some(array_data));
+        match (&self.event, &self.data) {
+            (MergedEvent::Dora(Event::Input { metadata, .. }), Some(data)) => {
+                let array = data
+                    .clone()
+                    .into_arrow_array(&metadata.type_info)
+                    .map_err(|err| arrow::pyarrow::PyArrowException::new_err(err.to_string()))?;
```

Comment:
```suggestion
                    .context("Could not create arrow array")?;
```
This enables to trace the error better.
E.g:
before
```bash

Traceback (most recent call last):
  File "/ros2-bridge/examples/python-ros2-dataflow/random_turtle.py", line 45, in <module>
    print(event["value"], flush=True)
          ~~~~~^^^^^^^^^
pyarrow.lib.ArrowException: Error creating Arrow Array

```
after
```
Traceback (most recent call last):
  File "/ros2-bridge/examples/python-ros2-dataflow/random_turtle.py", line 45, in <module>
    print(event["value"], flush=True)
          ~~~~~^^^^^^^^^
RuntimeError: Could not create array

Caused by:
   0: Error creating Arrow Array
   1: Invalid argument error: Expected 2 buffers in array of type Utf8, got 1


```

Reply:
Thanks!